src/main/java/com/owncloud/android/ui/trashbin/TrashbinActivity.java

```
@@ -82,7 +87,17 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        final User user = accountProvider.getUser();
+        user = accountProvider.getUser();
+        final String targetAccount = getIntent().getStringExtra(ARG_TARGET_ACCOUNT_NAME);
+        if (!user.getAccountName().equals(targetAccount)) {
+            final boolean accountSwitchSuccessful = getUserAccountManager().setCurrentOwnCloudAccount(targetAccount);
+            if (accountSwitchSuccessful) {
+                user = getUserAccountManager().getUser(targetAccount).orElse(user);
+            } else {
+                Log.w(TAG, "Tried to switch current account to \"" + targetAccount + "\", but could not be found.");
```

This should be logged via `Logger` API, although you are simply ignoring the error here, so the user will land in an unexpected place.

I think this is not good. Some sort of error pop up could be better maybe? Toast and quit?

---

Logging replaced with

> Toast and quit

as proposed